nixos/modules/system/boot/stage-1.nix

```diff
@@ -513,7 +513,13 @@ in
     };
 
     boot.initrd.compressor = mkOption {
-      default = "gzip";
+      default = (
+        if lib.versionAtLeast config.boot.kernelPackages.kernel.version "5.9" then
+          "${lib.getBin pkgs.zstd}/bin/zstd -10"
```

Comment:
The right way to do this after #97145 is just
```nix
{ 
  default = if lib.versionAtLeast config.boot.kernelPackages.kernel.version "5.9" then "zstd" else "gzip";
}
```

Reply:
Neat!